### DISCUSSION TITLE
[@types/react] Why is MouseEventHandler the default for onClick and not PointerEventHandler?

### DISCUSSION THREAD

**Discussion by thexpand** [DefinitelyTyped/DefinitelyTyped#61447] [Issues with a @types package]
IIRC, the `onClick` of a button binds both the mouse and touch events to it, e.g. I can use a mouse, my touch screen, and my stylus pen to "click" on an element when it has the `onClick` prop. However, TypeScript always infers that the type of the event handler is `MouseEventHandler`, which indicates only mouse devices.

I tried to learn more about the decision of why it is like it is, but couldn't find enough information anywhere. I would appreciate it if someone could explain this to me.

[a comment by an automated account is not shown]

**Comment by jbovone** [comment]
I haven't author any of the React types. 

But there is a thing with this, React has it's own synthetic events that doesn't necessarily correlate to the DOM counterparts.

It happens to me that, if you, as a developer, open up an onClick event listener you aren't as interested in the factual initiator of the event as much as the options that the interface provides to define certain behavior.    

Touch events have their own listeners, that behaves differently from click events.

**Comment by Jessidhia** [comment]
This is due to how React constructs the synthetic event internally. [`click` and other `click`-family events are treated as a `MouseEvent`](https://github.com/facebook/react/blob/05a55a4b09b7b7c8f63778fb8252a001ca66f8d7/packages/react-dom/src/events/plugins/SimpleEventPlugin.js#L92-L110), while only [events with `pointer` in the name are treated as `PointerEvent`](https://github.com/facebook/react/blob/05a55a4b09b7b7c8f63778fb8252a001ca66f8d7/packages/react-dom/src/events/plugins/SimpleEventPlugin.js#L146-L155).

`click` is also [documented to use a `MouseEvent` interface in MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event), though it is [documented to use `PointerEvent` in the spec](https://w3c.github.io/uievents/#event-type-click). It looks like this is a relatively recent change: the 2021-09-16 version of the spec has it as `PointerEvent`, but the previous published draft from 2019-05-30 still has it as `MouseEvent`.

FWIW, browsers do generate `click` events whenever an element is "activated", so it handles mouse, touchscreens, keyboard (say, tabbing to a button and pressing space), etc.

**Reply by thexpand** [reply]
That's interesting. Then, if the event generated from the browser handles all types of "click/tap/etc." events, doesn't it mean that the event contains all the properties of a `PointerEvent`?
